e2e_playwright/st_heading.py

```diff
@@ -22,11 +22,25 @@
 st.title("`Code` - Title with Anchor", anchor="title")
```

Comment:
The other header and title cases here do not seem to have a screenshot test. Is that planned to be added later?

Reply:
The original cypress test did not have screenshots for them, but good point, I will add it!

Reply:
Added more screenshot tests, including hovering over it to see the anchor button etc.